src/nix/get-env.sh

```diff
@@ -8,7 +8,13 @@ if [[ -n $stdenv ]]; then
     source $stdenv/setup
 fi
 
-for __output in $outputs; do
+if [ -e .attrs.sh ]; then
+    __olist="${!outputs[@]}"
+else
+    __olist=$outputs
+fi
+
+for __output in $__olist; do
```

Comment:
I wouldn’t mind seeing a test for this, because that’s way too many bash-isms for my brain to parse, so I’ve no idea how (and whether) it works